### ISSUE TITLE
AWS instance

### ISSUE THREAD

**Issue by Turabieh1982** [Turabieh1982/TestAWS#1]
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>AWS Instance Selector for Snowflake</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            line-height: 1.6;
            margin: 0;
            padding: 0;
            background-color: #f4f4f4;
        }
        header, footer {
            background-color: #333;
            color: white;
            text-align: center;
            padding: 1em 0;
        }
        main {
            padding: 2em;
            max-width: 800px;
            margin: auto;
            background: white;
            box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
        }
        h1, h2 {
            color: #333;
        }
        label {
            display: block;
            margin-bottom: 0.5em;
        }
        input, select {
            width: 100%;
            padding: 0.5em;
            margin-bottom: 1em;
            border: 1px solid #ccc;
            border-radius: 4px;
        }
        fieldset {
            border: 1px solid #ccc;
            padding: 1em;
            margin-bottom: 1em;
        }
        button {
            background-color: #28a745;
            color: white;
            border: none;
            padding: 1em 2em;
            cursor: pointer;
            border-radius: 4px;
        }
        button:hover {
            background-color: #218838;
        }
        #result {
            margin-top: 2em;
        }
        footer p {
            margin: 0;
        }
        table {
            width: 100%;
            border-collapse: collapse;
        }
        td {
            padding: 0.5em;
        }
    </style>
</head>
<body>
    <header>
        <h1>AWS Instance Selector</h1>
        <p>Optimize your Snowflake data processing with the best AWS instance</p>
    </header>

    <main>
        <form id="awsForm">
            
            <h3>Snowflake and ML/DL Questions</h3>
            <label for="snowflake-data-size">Snowflake Data Size (in GB):</label>
            <input type="number" id="snowflake-data-size" name="snowflake-data-size" required>

            <label>Are you using any machine learning or deep learning frameworks?</label>
            <table>
                <tr>
                    <td><input type="radio" id="use-ml" name="ml-dl" value="use-ml" required></td>
                    <td><label for="use-ml">Use Machine Learning</label></td>
                </tr>
                <tr>
                    <td><input type="radio" id="use-dl" name="ml-dl" value="use-dl" required></td>
                    <td><label for="use-dl">Use Deep Learning</label></td>
                </tr>
                <tr>
                    <td><input type="radio" id="use-both" name="ml-dl" value="use-both" required></td>
                    <td><label for="use-both">Use Both</label></td>
                </tr>
            </table>

            <label for="ml-framework">Please specify the machine learning or deep learning frameworks you are using:</label>
            <input type="text" id="ml-framework" name="ml-framework" placeholder="e.g., TensorFlow, PyTorch, scikit-learn">

            <label for="data-preprocessing">Do you perform extensive data preprocessing tasks (e.g., one-hot encoding, normalization)?</label>
            <select id="data-preprocessing" name="data-preprocessing" required>
                <option value="yes">Yes</option>
                <option value="no">No</option>
            </select>

            <label for="feature-selection">Do you use feature selection algorithms?</label>
            <select id="feature-selection" name="feature-selection" required>
                <option value="yes">Yes</option>
                <option value="no">No</option>
            </select>

            
            <h3>Required Questions</h3>
            <label for="networkPerformance">Required Network Performance:</label>
            <select id="networkPerformance" name="networkPerformance">
                <option value="low">Low</option>
                <option value="medium">Medium</option>
                <option value="high">High</option>
            </select>

            <label for="spotInstances">Use Spot Instances:</label>
            <select id="spotInstances" name="spotInstances">
                <option value="yes">Yes</option>
                <option value="no">No</option>
            </select>

            <label for="managedServices">Use Managed Services (e.g., SageMaker):</label>
            <select id="managedServices" name="managedServices">
                <option value="yes">Yes</option>
                <option value="no">No</option>
            </select>

            <label for="budgetConstraints">Budget Constraints (if any):</label>
            <input type="text" id="budgetConstraints" name="budgetConstraints">

            <label for="usageDuration">Expected Duration of Usage (in hours per day):</label>
            <input type="number" id="usageDuration" name="usageDuration">

            
            <h3>Additional Questions</h3>
            <label for="monthlyBudget">What is your monthly budget for the instance?</label>
            <input type="number" id="monthlyBudget" name="monthlyBudget">

            <label for="performanceOrMemory">Do you need high computational performance or more memory?</label>
            <select id="performanceOrMemory" name="performanceOrMemory">
                <option value="highPerformance">High Performance</option>
                <option value="moreMemory">More Memory</option>
            </select>

            <label for="scalability">Is scalability a key factor for your application?</label>
            <select id="scalability" name="scalability">
                <option value="yes">Yes</option>
                <option value="no">No</option>
            </select>

            <label for="mainPriority">What is the main priority for your application?</label>
            <input type="text" id="mainPriority" name="mainPriority">

            
            <fieldset>
                <legend>Instance Details</legend>
                <label for="computational-intensity">Computational Intensity:</label>
                <table>
                    <tr>
                        <td><input type="radio" id="cpu-intensive" name="computational-intensity" value="cpu-intensive" required></td>
                        <td><label for="cpu-intensive">CPU Intensive</label></td>
                    </tr>
                    <tr>
                        <td><input type="radio" id="memory-intensive" name="computational-intensity" value="memory-intensive" required></td>
                        <td><label for="memory-intensive">Memory Intensive</label></td>
                    </tr>
                    <tr>
                        <td><input type="radio" id="gpu-intensive" name="computational-intensity" value="gpu-intensive" required></td>
                        <td><label for="gpu-intensive">GPU Intensive</label></td>
                    </tr>
                </table>

                <label for="storage-type">Preferred Storage Type:</label>
                <select id="storage-type" name="storage-type" required>
                    <option value="ssd">SSD (faster and more reliable, but more expensive)</option>
                    <option value="hdd">HDD (slower but cheaper)</option>
                </select>
            </fieldset>

            
            <button type="submit">Submit</button>
            <button type="button" onclick="clearForm()">Clear Form</button>
        </form>

        <div id="result"></div>
    </main>

    <footer>
        <p>&copy; 2024 AWS Instance Selector</p>
    </footer>

    <script>
        function clearForm() {
            document.getElementById("awsForm").reset();
        }

        document.getElementById('awsForm').addEventListener('submit', function(event) {
            event.preventDefault();

            const computationalIntensity = document.querySelector('input[name="computational-intensity"]:checked').value;
            const performanceOrMemory = document.getElementById('performanceOrMemory').value;
            let instanceRecommendation = '';
            let costEstimate = '';

            if (computationalIntensity === 'cpu-intensive') {
                if (performanceOrMemory === 'highPerformance') {
                    instanceRecommendation = 'Sagemaker-CPUOptimal-XLarge';
                    costEstimate = '45.70 USD per month';
                } else if (performanceOrMemory === 'moreMemory') {
                    instanceRecommendation = 'Sagemaker-RAMOptimal-XLarge';
                    costEstimate = '51.52 USD per month';
                }
            } else if (computationalIntensity === 'memory-intensive') {
                if (performanceOrMemory === 'highPerformance') {
                    instanceRecommendation = 'Sagemaker-CPUOptimal-2XLarge';
                    costEstimate = '91.39 USD per month';
                } else {
                    instanceRecommendation = 'Sagemaker-RAMOptimal-2XLarge';
                    costEstimate = '103.26 USD per month';
                }
            } else if (computationalIntensity === 'gpu-intensive') {
                if (performanceOrMemory === 'highPerformance') {
                    instanceRecommendation = 'Sagemaker-GPUV100-2XLarge';
                    costEstimate = '856.80 USD per month';
                } else {
                    instanceRecommendation = 'Sagemaker-GPUAccel-2XLarge';
                    costEstimate = '210.56 USD per month';
                }
            }

            document.getElementById('result').innerHTML = `
                <h2>Recommended AWS Instance:</h2>
                <p><strong>Instance Type:</strong> ${instanceRecommendation}</p>
                <p><strong>Estimated Cost:</strong> ${costEstimate}</p>
            `;
        });
    </script>
</body>
</html>
